tests/integ/modin/index/test_index_methods.py

```diff
@@ -212,3 +219,173 @@ def test_df_index_shape(native_df):
+def test_index_item(native_index):
+    snow_index = pd.Index(native_index)
+    # 1 query to check length in item method, if length is 1, issue another query to get item
+    if len(native_index) == 1:
+        with SqlCounter(query_count=2):
+            assert snow_index.item() == native_index.item()
+    else:
+        with SqlCounter(query_count=1):
+            with pytest.raises(
+                expected_exception=ValueError,
+                match="can only convert an array of size 1 to a Python scalar",
+            ):
+                snow_index.item()
+
+
+@pytest.mark.parametrize("native_df", TEST_DFS)
+def test_df_index_columns_item(native_df):
+    snow_df = pd.DataFrame(native_df)
+    # 1 query to check length in item method, if length is 1, issue another query to get item
+    if len(native_df.index) == 1:
+        with SqlCounter(query_count=2):
+            assert snow_df.index.item() == native_df.index.item()
+    else:
+        with SqlCounter(query_count=1):
+            with pytest.raises(
+                expected_exception=ValueError,
+                match="can only convert an array of size 1 to a Python scalar",
+            ):
+                snow_df.index.item()
+
+    # no queries needed for columns
+    with SqlCounter(query_count=0):
+        if len(native_df.columns) == 1:
+            assert snow_df.columns.item() == native_df.columns.item()
+        else:
+            with pytest.raises(
+                expected_exception=ValueError,
+                match="can only convert an array of size 1 to a Python scalar",
+            ):
+                snow_df.columns.item()
+
+
+@sql_count_checker(query_count=1)
+@pytest.mark.parametrize("native_index", NATIVE_INDEX_TEST_DATA)
+def test_index_to_list(native_index):
+    snow_index = pd.Index(native_index)
+    assert_equal(native_index.to_list(), snow_index.to_list())
+
+
+@pytest.mark.parametrize("native_df", TEST_DFS)
+@sql_count_checker(query_count=1)
+def test_df_index_columns_to_list(native_df):
+    snow_df = pd.DataFrame(native_df)
+    assert_equal(native_df.index.to_list(), snow_df.index.to_list())
+    assert_equal(native_df.columns.to_list(), snow_df.columns.to_list())
+
+
+@sql_count_checker(query_count=3, join_count=1)
+@pytest.mark.parametrize("native_index", NATIVE_INDEX_TEST_DATA)
+def test_index_to_series(native_index):
+    print(native_index.to_series())
+    snow_index = pd.Index(native_index)
+    assert_series_equal(
+        snow_index.to_series(), native_index.to_series(), check_index_type=False
+    )
+    assert_series_equal(
+        native_index.to_series(index=range(10, 10 + len(native_index)), name="name"),
+        snow_index.to_series(index=range(10, 10 + len(snow_index)), name="name"),
+        check_index_type=False,
+    )
+
+
+@sql_count_checker(query_count=4, join_count=1)
+@pytest.mark.parametrize("native_df", TEST_DFS)
+def test_df_index_columns_to_series(native_df):
+    snow_df = pd.DataFrame(native_df)
```

Comment:
do we have any test case whose index actually has a name?

Reply:
I did not add that because we do not fully support index name api now